pd/src/app.rs

```diff
@@ -74,6 +74,9 @@ pub struct App {
 
     /// Used to allow asynchronous requests to be processed sequentially.
     completion_tracker: CompletionTracker,
+
+    /// Epoch duration in blocks
+    epoch_duration: Option<i64>,
```

Comment:
What does it mean if this is `None`?

Reply:
If it's `None` it means that the App has been instantiated but hasn't received an `InitChain` message yet.

Reply:
Is it important to ensure that we crash on app data that doesn't specify an epoch duration, or could we set a default one in the constructor and then overwrite it in the `InitChain` handler? If there's not a particular reason we don't want to have a default, it might be cleaner to know that the value is there so we don't have to constantly unwrap it.

Reply:
> Is it important to ensure that we crash on app data that doesn't specify an epoch duration, or could we set a default one in the constructor and then overwrite it in the InitChain handler?

I think I like the idea of specifying a default, but none of the other data within the `InitChain` `app_state` currently has a default. Should we also supply a default for the genesis notes in code?

Reply:
We can require that the field is present in the `InitChain` `app_state`, so we know we'll overwrite whatever default we set in the code (and the default in the code is just ensuring that we're not constantly option-unwrapping).  Or, we could let the other fields be optional and have default values. A reasonable default for the genesis notes would be an empty list, although this isn't useful for setting up a chain.